File: CPP/modelo/AccionColocarTorre.cpp
#ifndef ACCION_COLOCAR_TORRE_CPP
#define ACCION_COLOCAR_TORRE_CPP

#include "Torre.cpp"

class AccionColocarTorre {
private:
    Torre* torre;
    int indiceEnLista;

public:
    AccionColocarTorre(Torre* torre, int indiceEnLista) {
        this->torre = torre;
        this->indiceEnLista = indiceEnLista;
    }

    Torre* getTorre() const { return torre; }
    int getIndiceEnLista() const { return indiceEnLista; }
};

#endif

File: Java/src/modelo/AccionColocarTorre.java
package modelo;

public class AccionColocarTorre {
    private final Torre torre;
    private final int indiceEnLista;

    public AccionColocarTorre(Torre torre, int indiceEnLista) {
        this.torre = torre;
        this.indiceEnLista = indiceEnLista;
    }

    public Torre getTorre() { return torre; }
    public int getIndiceEnLista() { return indiceEnLista; }
}
